### DISCUSSION TITLE
Boom offsets

### DISCUSSION THREAD

**Discussion by freggala** [Open-Agriculture/AgIsoStack-plus-plus#314] [General]
In the task_controller_client example there are three DeviceProperties defined
```
retVal &= poolToPopulate->add_device_property("Offset X", 0, static_cast<std::uint16_t>(isobus::DataDescriptionIndex::DeviceElementOffsetX), static_cast<std::uint16_t>(ImplementDDOPObjectIDs::ShortWidthPresentation), static_cast<std::uint16_t>(ImplementDDOPObjectIDs::BoomXOffset));
	retVal &= poolToPopulate->add_device_property("Offset Y", 0, static_cast<std::uint16_t>(isobus::DataDescriptionIndex::DeviceElementOffsetY), static_cast<std::uint16_t>(ImplementDDOPObjectIDs::ShortWidthPresentation), static_cast<std::uint16_t>(ImplementDDOPObjectIDs::BoomYOffset));
```
I would assume that changing the second argument of each of these calls to add_device_property(...) to a reasonable number (say e.g. 5000) should shift the longitudinal and lateral position of the Boom relative to its connection point. However, it just doesn't. How is it supposed to be done correctly?

**Comment by ad3154** [comment]
In theory that should work, but one thing to keep in mind is that if you've previously uploaded a DDOP to this same TC, the new offset will not be used unless you also change the structure label on the DDOP, because that is what primarily controls if the DDOP is re-uploaded.

You could also maybe change it to a DPD in theory, and send a value command to the TC instead, again in theory, or have the TC request it by adding it to the default set (I think), but that will probably be less robust than using a property.

Keep in mind as well that the ISO coordinate system defines Y as the left to right offset (which feels weird) relative to the device reference point, and the X offset is the fore/aft offset. The reference point of the vehicle is the center of the rear axle generally, to which you reference the connector, to which you again reference the implement.

If it still doesn't work after forcing the DDOP to re-upload, then we can investigate other structural things that might be going on in the DDOP.
